pkg/virt-controller/services/template.go

```diff
@@ -1326,21 +1327,28 @@ func (t *templateService) RenderHotplugAttachmentPodTemplate(volume *v1.Volume,
 	return pod, nil
 }
 
-func getRequiredCapabilities(vmi *v1.VirtualMachineInstance) []k8sv1.Capability {
-	res := []k8sv1.Capability{}
+func getRequiredCapabilities(vmi *v1.VirtualMachineInstance, config *virtconfig.ClusterConfig) []k8sv1.Capability {
+	capabilities := []k8sv1.Capability{}
+
 	if (len(vmi.Spec.Domain.Devices.Interfaces) > 0) ||
 		(vmi.Spec.Domain.Devices.AutoattachPodInterface == nil) ||
 		(*vmi.Spec.Domain.Devices.AutoattachPodInterface == true) {
-		res = append(res, CAP_NET_ADMIN)
+		capabilities = append(capabilities, CAP_NET_ADMIN)
 	}
 	// add a CAP_SYS_NICE capability to allow setting cpu affinity
-	res = append(res, CAP_SYS_NICE)
+	capabilities = append(capabilities, CAP_SYS_NICE)
 
 	// add CAP_SYS_ADMIN capability to allow virtiofs
 	if util.IsVMIVirtiofsEnabled(vmi) {
-		res = append(res, CAP_SYS_ADMIN)
+		capabilities = append(capabilities, CAP_SYS_ADMIN)
 	}
-	return res
+
+	// add CAP_SYS_RESOUCE capacility to allow Live Migration for VM with SRIOV interfaces
+	if config.SRIOVLiveMigrationEnabled() && util.IsSRIOVVmi(vmi) {
```

Comment:
do we have any concern that someone could attempt to start the VMI with SRIOV, then flip this feature gate on afterwards? When that occurs, the VMI that launched before the feature gate is set will not be live migratable.

I think your logic as it is now is the right thing to do. Maybe worth simply documenting this edge case so people are aware.

Reply:
I think it will actually be migratable, as this affects the target node (when the pod there is created, it should have the right capability in order to allow the hot-plug).

There is some corner cases where this will cause some issues: One of the stages is to try and revert the detached devices on the source in case the migration fails (i.e. attach them back to the domain). That one will not work if I follow your scenario correctly.

Reply:
In more details , currently switching the feature-gate on affects the target pod creation stage of the migration, where it calls [`renderLaunchManifest`](https://github.com/kubevirt/kubevirt/blob/master/pkg/virt-controller/services/template.go#L318) that conditionally add the required capability according to the feature-gate state (the mentioned line).
If you have running VM with SRIOV  you want to migrate when the feature-gate is on, the target VM pod will have the required capability. 
So after the migration is finished attaching the SRIOV devices at the target should not fail.

It was also suggested to block migration for VM's with SRIOV by default if the feature gate is off
https://github.com/kubevirt/kubevirt/pull/4874#pullrequestreview-575894290